web/frontend/src/components/gamepad/index.svelte

```diff
@@ -0,0 +1,328 @@
+<script lang='ts'>
+
+import { grpc } from '@improbable-eng/grpc-web';
+import { onMount, onDestroy } from 'svelte';
+import { Timestamp } from 'google-protobuf/google/protobuf/timestamp_pb';
+import { ConnectionClosedError } from '@viamrobotics/rpc';
+import {
+    Client,
+    inputControllerApi as InputController,
+    type ResponseStream,
+    robotApi,
+    type ServiceError,
+} from '@viamrobotics/sdk';
+import { toast } from '../../lib/toast';
+import { rcLogConditionally } from '../../lib/log';
+
+export let name:string;
+export let client:Client;
+export let statusStream: ResponseStream<robotApi.StreamStatusResponse> | null;
+    
+let gamepadIdx = <number | null> (null);
+let gamepadConnectedPrev = false;
+let enabled = false;
+    
+const curStates = <Record<string, number>>({
+    X: Number.NaN,
+    Y: Number.NaN,
+    RX: Number.NaN,
+    RY: Number.NaN,
+    Z: Number.NaN,
+    RZ: Number.NaN,
+    Hat0X: Number.NaN,
+    Hat0Y: Number.NaN,
+    South: Number.NaN,
+    East: Number.NaN,
+    West: Number.NaN,
+    North: Number.NaN,
+    LT: Number.NaN,
+    RT: Number.NaN,
+    LThumb: Number.NaN,
+    RThumb: Number.NaN,
+    Select: Number.NaN,
+    Start: Number.NaN,
+    Menu: Number.NaN,
+});
+    
+let handle = -1;
+let prevStates: Record<string, number> = {};
+
+let lastError = Date.now();
+const sendEvent = (newEvent: InputController.Event) => {
+    if (!enabled) {
+    return;
+    }
+    const req = new InputController.TriggerEventRequest();
+    req.setController(name);
+    req.setEvent(newEvent);
+    rcLogConditionally(req);
+    client.inputControllerService.triggerEvent(req, new grpc.Metadata(), (error: ServiceError | null) => {
+    if (error) {
+        if (ConnectionClosedError.isError(error)) {
+        return;
+        }
+        const now = Date.now();
+        if (now - lastError > 1000) {
+        lastError = now;
+        toast.error(error.message);
+        }
+    }
+    });
+};
+    
+let lastTS = Timestamp.fromDate(new Date());
+const nextTS = () => {
+    let nowTS = Timestamp.fromDate(new Date());
+    if (lastTS.getSeconds() > nowTS.getSeconds() ||
+    (lastTS.getSeconds() === nowTS.getSeconds() && lastTS.getNanos() > nowTS.getNanos())) {
+    nowTS = lastTS;
+    }
+    if (nowTS.getSeconds() === lastTS.getSeconds() &&
+    nowTS.getNanos() === lastTS.getNanos()) {
+    nowTS.setNanos(nowTS.getNanos() + 1);
+    }
+    lastTS = nowTS;
+    return nowTS;
+};
+    
+const currentGamepad = () => {
+    return gamepadIdx === null ? null : navigator.getGamepads()[gamepadIdx];
+};
```

Comment:
This can be refactored to use svelte's `$` syntax:
```suggestion
$: currentGamepad = gamepadIdx === null ? null : navigator.getGamepads()[gamepadIdx];

Then in the templating section we can just do stuff like: 
```svelte
{#if currentGamepad?.connected}